### PR TITLE
Adding Perfectscale as Validated Partner addon

### PR DESCRIPTION
**Pull request by oveleten**
This PR will add Perfectscale as partner addon.

It requires secrets to be added to AWS Secret Manager, but our AWS Partner hasn't added yet, maybe you can help?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shapirov103** [comment]
@oveleten please send an email to elamaras@amazon.com and shapirov@amazon.com. Also please indicate if you are currently on the AWS Developers Slack workspace and whether you would like to be onboarded for faster feedback.

**Comment by oveleten** [comment]
@shapirov103 I sent mail . Thanks

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@oveleten Thankyou so much for PR. We cant proceed with this PR validation without the following:
1. Secrets to be shared to us in AWS channels.
2. A Test Job which functionally validates the functionality of our ISV product.

**Comment by oveleten** [comment]
@elamaran11 I sent secret to you and added test job, check please, thanks

**Comment by elamaran11** [comment]
@oveleten Please see the below logs. The test job is failing as it is checking on in correct endpoint for exporter pod. Also health checking a pod does not qualify for a functional test job. We need a functional test job which validates the functionality of your product. Please rework on the test job and send us an updated commit 

```bash
❯ k get all -n perfectscale                                                                                         ─╯
NAME                                                                READY   STATUS    RESTARTS   AGE
pod/perfectscale-perfectscale-exporter-778858786-9qzl2              1/1     Running   0          2m23s
pod/perfectscale-perfectscale-kube-state-metrics-7fccc94574-pj7rq   1/1     Running   0          2m23s
pod/perfectscale-test-job-nvksc                                     1/1     Running   0          13s

NAME                                                   TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
service/perfectscale-perfectscale-exporter             ClusterIP   10.101.166.107   <none>        80/TCP     2m23s
service/perfectscale-perfectscale-kube-state-metrics   ClusterIP   10.105.17.60     <none>        8080/TCP   2m23s

NAME                                                           READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/perfectscale-perfectscale-exporter             1/1     1            1           2m23s
deployment.apps/perfectscale-perfectscale-kube-state-metrics   1/1     1            1           2m23s

NAME                                                                      DESIRED   CURRENT   READY   AGE
replicaset.apps/perfectscale-perfectscale-exporter-778858786              1         1         1       2m23s
replicaset.apps/perfectscale-perfectscale-kube-state-metrics-7fccc94574   1         1         1       2m23s

NAME                              COMPLETIONS   DURATION   AGE
job.batch/perfectscale-test-job   0/1           13s        13s

░▒▓    /home/eksadmin ▓▒░
❯ k logs perfectscale-test-job-nvksc -n perfectscale                                                                ─╯
Service is not available yet. Retrying in 10 seconds...
Service is not available yet. Retrying in 10 seconds...
Service is not available yet. Retrying in 10 seconds...
Service is not available yet. Retrying in 10 seconds...
Service is not available yet. Retrying in 10 seconds...
```

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@oveleten Thanks for Secrets. The test job is failing and this health check job does not qualify for a functional test job.

File: eks-anywhere-common/Testers/Perfectscale/perfectscale-test-job.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  template:
+    spec:
+      containers:
+      - name: perfectscale-test-job
+        image: curlimages/curl:latest
+        command: ["sh", "-c"]
+        args:
+        - |
+          while true; do
+            if curl --silent --fail perfectscale-exporter:80/health ; then
```
[the remark this answers is not in the document]
Comment (oveleten) [reply eks-anywhere-common/Testers/Perfectscale/perfectscale-test-job.yaml]: I fixed test job, can you run again?

[the remark this answers is not in the document]
Comment (elamaran11) [reply eks-anywhere-common/Testers/Perfectscale/perfectscale-test-job.yaml]: @oveleten This health check job does not qualify for a functional test job. Please check [here](https://github.com/aws-samples/eks-anywhere-addons#:~:text=Though%20deployment%20and,Example%20here.) for more information on expectations for functional test job. Here is an [approved functional test job](https://github.com/aws-samples/eks-anywhere-addons/blob/main/eks-anywhere-common/Testers/Hashicorp/Vault/kvJob.yaml) for your reference.

[the remark this answers is not in the document]
Comment (oveleten) [reply eks-anywhere-common/Testers/Perfectscale/perfectscale-test-job.yaml]: I rewrote job  with logic that check if our exporter was able to connect to the backed and send data via API call

**Review by elamaran11 — COMMENTED** [review]
@oveleten I was able to validate Perfectscale with the functional job in vSphere and Baremetal environment successfully. We will get to Snow and LocalCluster by start of next week. Thanks for your patience with us.

**Comment by elamaran11** [comment]
@oveleten Perfectscale addon works fine in EKS-A on Snowball today. Only pending deployment model is EKS Localclusters which we will get to by this week.

**Comment by shapirov103** [comment]
@oveleten @elamaran11 validated on Rover (EKS on Outposts) - no failures

**Review by elamaran11 — APPROVED** [review]
@oveleten We have completed validation across all 4 deployment models - vSphere, BareMetal, Snowball, Local Clusters. We will merging this PR Soon.

**Pull request merged**